docs/docs/crafty.md

```diff
@@ -3,4 +3,184 @@ id: crafty
+
+## Contracts
+
+Crafty works with only two contracts:
+* [`Crafty`](https://github.com/zeppelinos/crafty/blob/master/contracts/Crafty.sol) itself, containing the game logic (how new recipes are added, how crafting takes place, etc.) plus some role-based access control ([RBAC](https://github.com/OpenZeppelin/openzeppelin-solidity/blob/zos-release/contracts/ownership/rbac/RBAC.sol), powered by [OpenZeppelin](https://openzeppelin.org/)), for admin tasks, such as deletion of tokens.
+* The [`CraftableToken`](https://github.com/zeppelinos/crafty/blob/master/contracts/CraftableToken.sol)s, which are a simple extension of a mintable ERC20, holding token creator data, a URI for metadata, and a list of required ingredients, along with the amounts for each.
+
+To better see how these two contracts interact, take a look at `Crafty`'s `craft` method, where a user provides the address of a `CraftableToken` to craft, and `Crafty` makes sure the required ingredients are consumed before minting it.
+
+```
+function craft(CraftableToken _craftable) public {
+  address player = msg.sender;
+
+  uint256 totalSteps = _craftable.getTotalRecipeSteps();
+  for (uint i = 0; i < totalSteps; ++i) {
+    ERC20 ingredient;
+    uint256 amountNeeded;
+    (ingredient, amountNeeded) = _craftable.getRecipeStep(i);
+
+    ingredient.transferFrom(player, address(this), amountNeeded);
+  }
+
+  // Issue the crafted token - for ingredient-less tokens, a higher number of
+  // tokens are minted.
+  uint256 tokensToMint = (totalSteps == 0) ? 100 : 1;
+  _craftable.mint(player, tokensToMint);
+}
+```
+
+## Developing with ZeppelinOS
+
+So, how would we go about integrating ZeppelinOS to such a project? The process is remarkably simple: we'll cover it step by step. But before we get started, don't forget to initialize your project with ZeppelinOS:
+
+```sh
+zos init Crafty
+```
+
+This will create a `zos.json` file, which will store how your project is structured in ZeppelinOS: you'll probably want to commit this file.
+
+### Making `Crafty` upgradeable
+
+There are multiple reasons why we would want to be able to upgrade `Crafty`'s game logic contract: from bug-fixing, to adding new functionalities. Remember how `Crafty` uses RBAC (role-based access control)? An upgrade could easily add new roles, such as a `curator` role, which would be in charge of approving new tokens before they are added to the game, or highlighting featured creations to be displayed on the front page. These simple but useful extensions would not be possible without ZeppelinOS.
+
+`Crafty` is a great example of how easy it is to add upgradeability to your project. It extends OpenZeppelin contracts (the `RBAC` contract), which are safe to use with ZeppelinOS, so the only change that we need to make is a minor one: the constructor must be removed, and its code added to an `initialize` function:
+
+```
+import 'zos-lib/contracts/migrations/Initializable.sol';
+
+contract Crafty is Initializable {
```

Comment:
AFAIK, `Initializable` where possible to avoid having to introduce too many concepts. `CraftableToken` inherits from `Migratable` contracts, so those are mentioned in this doc.